compiler/env/jittypes.h

```diff
@@ -149,7 +149,9 @@ typedef struct TR_InlinedCallSite
       #define FLUSH_MEMORY(smp)
    #endif
 #else
-   #if (__IBMC__ || __IBMCPP__)
+   #ifdef __ibmxl__
+      #define FLUSH_MEMORY(smp) if ( smp ) __lwsync();
+   #elif (__IBMC__ || __IBMCPP__)
       void do_lwsync (void);
       #pragma mc_func do_lwsync {"7c2004ac"}
       #pragma reg_killed_by do_lwsync
```

Comment:
I think both old and new XL support the `__lwsync()` builtin, so can you try getting rid of the `mc_func` part and using the same define for both?